tests/rptest/tests/compatibility/arroyo_test.py

```diff
@@ -0,0 +1,69 @@
+# Copyright 2022 Redpanda Data, Inc.
+#
+# Use of this software is governed by the Business Source License
+# included in the file licenses/BSL.md
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0
+
+import os
+import subprocess
+import contextlib
+
+from rptest.services.cluster import cluster
+from rptest.tests.redpanda_test import RedpandaTest
+
+
+@contextlib.contextmanager
+def set_env(**env):
+    previous_env = dict(os.environ)
+    os.environ.update(env)
+    try:
+        yield
+    finally:
+        os.environ.clear()
+        os.environ.update(previous_env)
+
+
+class ArroyoTest(RedpandaTest):
+    """
+    Run the arroyo test suite against a redpanda cluster in
+    a ducktape environment.
+
+    The test suite lives here under tests/: https://github.com/getsentry/arroyo.
+    """
+    TEST_SUITE_PATH = "/root/external_test_suites/arroyo"
+
+    def _find_failed_tests(self, pytest_output: list[bytes]):
+        return [line for line in pytest_output if b"FAILED" in line]
+
+    @cluster(num_nodes=3)
+    def test_arroyo_test_suite(self):
+        try:
+            with set_env(DEFAULT_BROKERS=self.redpanda.brokers()):
+                result = subprocess.run([
```

Comment:
This line will run the arroyo tests on the orchestration node. Typically, whenever a test case requires auxiliary apps or programs, we run them on another ducktape node. You can easily grab a ducktape node using a [BackgroundThreadService](https://github.com/confluentinc/ducktape/blob/master/ducktape/services/background_thread.py#L23).

For an example of a BackgroundThreadService, look at the `RpkConsumer` in bytes_sent_test.py:
```
    def _consume_and_count_bytes(self):
        consumer = RpkConsumer(self._ctx, self.redpanda, self.topic)
        consumer.start()

        self._bytes_received = 0

        def count_bytes():
            for msg in consumer.messages:
                value = msg["value"]

                # Ignore None values
                if value is None:
                    return False

                self._bytes_received += len(value)

            # Returns true the first time any bytes
            # are fetched.
            return self._bytes_received > 0
 ```

Reply:
Or the test could inherit from [PreallocNodesTest](https://github.com/redpanda-data/redpanda/blob/de6bcdb6e196e274ccc7532e6658599739e2daab/tests/rptest/tests/prealloc_nodes.py#L16) and simply invoke the pytest command on the allocated node. The main point here is to avoid running a separate application on the orchestration node.

So something like:
```
node = self.preallocated_nodes[0]
for line in node.account.ssh_capture(f'python3 -m pytest {TEST_SUITE_PATH} -k KafkaStreamsTestCase -rf'):
     # .. you can parse the line here
```

Reply:
Also, thank you for jumping on a call with me and explaining how this stuff works.
The initial comment is addressed, so I'll resolve but feel free to re-open if needed.

Reply:
not a problem